### PR TITLE
[PR] Create `Dart` package

### LINKED ISSUE #10: Question: How are we going to use statuses in `Dart`/`Flutter`?
As noted in https://github.com/dwyl/statuses/issues/4#issuecomment-1179609113 I feel that **`JSON`** is a more versatile format for storing the `statuses` list than a `List` of `Elixir` `Map` or `Struct` ... 

As per https://github.com/dwyl/product-roadmap/issues/40 we are building a **`Flutter`** (`Native` Mobile App) version of the **`MVP`** that will be **_offline_ first**. We need `statuses` in `Dart`-land so that it can be included in the _compiled_ `Native` App. 

We could _either_ create a [`Struct`](https://api.flutter.dev/flutter/dart-ffi/Struct-class.html) in `Dart` and _duplicate_ the `statuses` 
and thus have to _maintain_ them in multiple places ... **`!DRY`**

***or***

use **`JSON`**.

### PR DESCRIPTION
**Pull request by LuchoTurtle**
closes #10 

This creates a `Dart` package (that is already published in https://pub.dev/packages/statuses) and changes the CI workflows to take into account both `Elixir` and `Dart` packages.

The `Elixir` package fetches information from the `json` instead of a list of structs that are hardcoded. Both packages use the `statuses.json` file.

### CODE REVIEW

File: lib/statuses.ex [hunk 1]
```diff
@@ -5,113 +5,41 @@ defmodule Statuses do
...
+
+    # we need this `cd` to locate the file in `/deps`
+    case cwd =~ "/statuses" do
+      true ->
+        File.read!("statuses.json")
+        |> Jason.decode!()
+
+      # coveralls-ignore-start
+
+      # temporarily `cd` into `deps/quotes` dir and read `quotes.json` file:
```
Comment (nelsonic) [inline lib/statuses.ex:29]: This appears to have been copy-pasted from `quotes`. 
Consider updating to be correct with the `statuses` project. 💭

Comment (LuchoTurtle) [reply lib/statuses.ex]: Sorry about that. Should be fixed in [`5afc679` (#21)](https://github.com/dwyl/statuses/pull/21/commits/5afc679899c0e80f10eb46067230370e0c4bd480).

**Comment by LuchoTurtle** [comment]
Should be fixed in [5afc679 (#21)](https://github.com/dwyl/statuses/pull/21/commits/5afc679899c0e80f10eb46067230370e0c4bd480).

**Comment by LuchoTurtle** [comment]
The tests work because the path that was changed was "in case the `statuses.json` file didn't exist. However, this probably will never happen because the `json` file is shipped with the package. The pattern matching is just a "safe-guard", hence why it's also difficult to test the function. How can we test "the file is not there"? The only way to test is this is to pass the path as the function parameter. But I went against that because I wanted to be as similar to `dwyl/quotes` as possible so it'd be easier for people to understand.

**Review by nelsonic — APPROVED** [review]
@LuchoTurtle Looks good. Thanks. 🎉

**Pull request merged**